### ISSUE TITLE
WebSocket handshake failure when app is built with Bazel

### ISSUE THREAD

**Issue by luispadron** [apple/swift-nio#2599]
### Expected behavior

WebSocket handshake is successful between build systems using the same code.

### Actual behavior

Im experiencing some strange runtime behavior differences between a macOS app built with Xcode and the same app built with Bazel, would love some ideas.

The issue is were trying to create a web socket connection between a macOS app and an iOS client. The handshake for this works when the macOS app is built with Xcode (or SPM) but the handshake fails when I run the app built with Bazel (using [macos_application](https://github.com/bazelbuild/rules_apple/blob/master/doc/rules-macos.md#macos_application)).

The code is the exact same between the two build systems (for both the client and server). 
I've even double checked entitlements and Info.plist is all equivalent. 

The handshake fails because `Sec-WebSocket-Key` is not the expected hash. 

Here is what we see as a difference when trying to make the handshake, all other headers are the same between the two.

Xcode: `Sec-WebSocket-Key: DOwV4Cw/4Dk6ra1NiEVyrw==`
Bazel: `Sec-WebSocket-Key: B93lsr3zQ0p+Dx4Ymn62VQ==`

And the actual error message displayed on the iOS client:

```
nw_ws_validate_server_response  server response doesn't match expected value
Task <30115C27-C67C-4F9A-9D37-E6E1DD08C00E>.<1> finished with error [-1,011] Error Domain=NSURLErrorDomain Code=-1011 "There was a bad response from the server." UserInfo={NSErrorFailingURLStringKey=ws://127.0.0.1:34143/channel, NSErrorFailingURLKey=ws://127.0.0.1:34143/channel, _NSURLErrorWebSocketHandshakeFailureReasonKey=5, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalWebSocketTask <30115C27-C67C-4F9A-9D37-E6E1DD08C00E>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalWebSocketTask <30115C27-C67C-4F9A-9D37-E6E1DD08C00E>.<1>, NSLocalizedDescription=There was a bad response from the server.}
```

Any ideas what could be causing this difference? Are there any compile flags / defines we need to enable to make this work properly?

### SwiftNIO version/commit hash

`2.61.1`

### System & version information

macOS: 13.6.1 (arm64)
Xcode: 15.0.0
Bazel: 6.4.0
Swift: `swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1`

**Comment by Lukasa** [comment]
The websocket key (in `Sec-Websocket-Key`) is set by the client, not the server. It's typically generated randomly, but the server has no influence over it: it can only compute over it. Is it the case that the client repeatedly connect to the Xcode-built server, but not the Bazel-built server? Does it always fail to connect to the Bazel server?

**Comment by luispadron** [comment]
@Lukasa correct, the client is able to always connect to the server (macOS app) when it is built with Xcode but handshake always fails when I build the server with Bazel

**Comment by Lukasa** [comment]
Does this reproduce with the sample servers in this repo?

**Comment by luispadron** [comment]
I can give that a try and report back. 

I did try to create a minimal server and client to share but that worked for both build systems (this was using a stripped down version of both the server and client). 

The minimal server I built also worked with our large iOS client to establish a connection using both build systems. 

Is there anything I can enable to debug further? All I have to go off on is the error message I posted above. 

The handshake failure comes right after calling `resume` on a `URLSessionWebSocketTask` on the iOS client.

**Comment by Lukasa** [comment]
The other thing you can try to do to debug this is to get a packet capture of the two different handshakes.

**Comment by luispadron** [comment]
@Lukasa thanks for your help here! I'm very much a newb here. 

I haven't had a chance to get the exact web info yet but on the android client when it connects to the Bazel built server we do get some more info:

```
Throwable java.net.ProtocolException: Expected 'Sec-WebSocket-Accept' header value 'eabW+NOBcvdzRVggc2z4oTgL8Og=' but was 'CBn5S5K8ShGNaOtKxZc7/ldcyno='
```

**Comment by luispadron** [comment]
I've confirmed the root cause is `BYTE_ORDER` is coming back as `BIG_ENDIAN` which is incorrect for macOS arm64. Seems like a similar issue to: https://github.com/apple/swift-nio/pull/2584

**Comment by luispadron** [comment]
This minimal repo demonstrates the issue: https://github.com/luispadron/bazel-big-endian-on-macos

**Comment by Lukasa** [comment]
Can you confirm that you're using version 2.62.0? That contains the fix from #2584.

**Comment by luispadron** [comment]
Confirmed that 2.62.0 fixes this issue for Bazel build. Thanks so much for your responsiveness here @Lukasa
